### DISCUSSION TITLE
[SSE Subscriptions] What are the implications of using ponyfill?

### DISCUSSION THREAD

**Discussion by cabello** [trpc/trpc#6295] [Q&A]
From reading this discussion: https://github.com/trpc/trpc/discussions/6071
And also the docs https://trpc.io/docs/client/links/httpSubscriptionLink#custom-headers-through-ponyfill that starts with:
> Recommended for non-web environments

It seems that the **ideal** case is to use cookies for auth and benefit from SSE completely, but what if I use authorization tokens like this in my web environment:

```typescript
const subscriptionLink = unstable_httpSubscriptionLink({
  url: `${env.apiBaseUrl()}trpc`,
  transformer: superjson,
  EventSource: EventSourcePolyfill,
  eventSourceOptions: () => {
    const token = getUserToken()
    return {
      headers: {
        authorization: `Bearer ${token}`,
      },
    }
  },
})
```

What is the trade-off for doing this for web clients, what are we losing in this case?

Thank you for the clarification 🙏

**Comment by KATT** [comment]
If you're on the web, nothing beats cookies for auth, so I'd recommend trying to use cookies.

> What is the trade-off for doing this for web clients, what are we losing in this case?

I think you mainly lose the event source thing in chrome's inspector
